service/dap/server.go

```diff
@@ -1369,16 +1489,29 @@ func (s *Server) onSetFunctionBreakpointsRequest(request *dap.SetFunctionBreakpo
 	s.sendNotYetImplementedErrorResponse(request.Request)
 }
 
-// onStepBackRequest sends a not-yet-implemented error response.
-// Capability 'supportsStepBack' is not set 'initialize' response.
+// onStepBackRequest performs a ReverseNext command call, returning a NextResponse.
 func (s *Server) onStepBackRequest(request *dap.StepBackRequest) {
-	s.sendNotYetImplementedErrorResponse(request.Request)
+	isRecording, _ := s.debugger.Recorded()
+	if !isRecording {
+		s.sendErrorResponse(request.Request, UnableToContinue, "Unsupported command",
+			fmt.Sprintf("cannot process '%s' request, target is not a recording", request.Command))
+	}
+	s.send(&dap.StepBackResponse{Response: *newResponse(request.Request)})
+	s.doCommand(api.ReverseNext)
 }
 
-// onReverseContinueRequest sends a not-yet-implemented error response.
-// Capability 'supportsStepBack' is not set 'initialize' response.
+// onReverseContinueRequest performs a rewind command call up to the previous
+// breakpoint or the frame zero of the program, (dependant on Rewind/rr implementation)
 func (s *Server) onReverseContinueRequest(request *dap.ReverseContinueRequest) {
-	s.sendNotYetImplementedErrorResponse(request.Request)
+	isRecording, path := s.debugger.Recorded()
```

Comment:
This check is redundant, the backend already does it. Also we don't show the relative controls anymore if the backend isn't rr so it should never happen on a well behaved client.